include/vsg/vsg.hpp

```diff
@@ -182,15 +201,38 @@ struct VirtualNode : public VirtualNodeI<ContextT> {
     // Provide the base interface elements for our types.
     [[nodiscard]] auto markerAsAny() const -> std::any override { return std::any(m_changeMarker); }
     [[nodiscard]] auto drawArgsAsAny() const -> std::any override { return std::any(m_drawArgs); }
+    [[nodiscard]] auto drawArgsType() const -> std::type_info const & override {
+        return typeid(DrawArgsT);
+    }
     [[nodiscard]] auto children() const -> Vector<Box<VirtualNodeI<ContextT>>> override {
         return m_children;
     }
+    [[nodiscard]] auto hasProvidedChangeMarker() const -> bool override {
+        return typeid(m_changeMarker) != typeid(NoChangeMarkerProvided);
+    }
     [[nodiscard]] auto changeMarkerEquals(NodeI<ContextT> const &other) const -> bool override {
         auto otherChangeMarker = other.markerAsAny();
         if (otherChangeMarker.type() != typeid(m_changeMarker)) { return false; }
         return std::any_cast<ChangeMarkerT>(otherChangeMarker) == m_changeMarker;
     }
 
+    [[nodiscard]] auto drawArgsEquals(NodeI<ContextT> const &other) const -> bool override {
+        if constexpr (std::same_as<NoChangeMarkerProvided, ChangeMarkerT> && !std::equality_comparable<DrawArgsT>) {
+            static_assert(
+                std::equality_comparable<DrawArgsT>,
+                "The Draw Args you provided must be comparable with the == operator or you must "
+                "provide a change marker."
+            );
+        } else if constexpr (!std::same_as<NoChangeMarkerProvided, ChangeMarkerT> && !std::equality_comparable<DrawArgsT>) {
+            return false;
```

Comment:
I don't actually know what _should_ happen here, but I suppose returning false makes as much sense as anything else. Should this ever be called if a change marker was provided? If not, maybe an error instead of returning false might be better

Reply:
What kind of error? throwing an exception?